File: frontend/src/types/dto/entity/game/GameSettingsDto.ts
import type { Difficulty } from "@/types/enum/Difficulty";
import type { GameMode } from "@/types/enum/GameMode";
import type { GameType } from "@/types/enum/GameType";
import type { TimeLimitPreset } from "@/types/enum/TimeLimitPreset";

export type GameSettingsDto = {
    difficulty: Difficulty,
    timeLimit: TimeLimitPreset,
    gameMode: GameMode,
    gameType: GameType
}

File: backend/src/main/java/com/github/ryand6/sudokuweb/dto/entity/game/GameSettingsDto.java
package com.github.ryand6.sudokuweb.dto.entity.game;

import com.github.ryand6.sudokuweb.enums.Difficulty;
import com.github.ryand6.sudokuweb.enums.GameMode;
import com.github.ryand6.sudokuweb.enums.GameType;
import com.github.ryand6.sudokuweb.enums.TimeLimitPreset;
import lombok.*;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class GameSettingsDto {

    private Difficulty difficulty;

    private TimeLimitPreset timeLimit;

    private GameMode gameMode;

    private GameType gameType;

}
